### ISSUE TITLE
vanilla Rancher HA install: Catalogs all report Error in HTTP GET fetching from repos

### ISSUE THREAD

**Issue by sdanieru** [rancher/rancher#22155]
**What kind of request is this (question/bug/enhancement/feature request):**
BUG

**Steps to reproduce (least amount of steps as possible):**
*Follow Rancher HA install instructions found here:
https://rancher.com/docs/rancher/v2.x/en/installation/ha/
*For SSL configuration, follow the instructions for the default 'Rancher Generated Certificates'
*for hostname, provide `rancher.my.domain`
*After installation, navigate in UI and provide initial settings, including `https://rancher.my.domain`
*Navigate to Global > Tools > Catalogs
*Enable all three Catalogs: 'Library', 'Helm Stable', and 'Helm Incubator'

**Result:**
All of the Catalogs show an error message similar this:

`Error in HTTP GET to [https://git.rancher.io/charts/index.yaml], error: Get https://git.rancher.io/charts/index.yaml: x509: certificate is valid for *.my.domain, my.domain, not git.rancher.io`

**Other details that may be helpful:**
Initially I tried this install with 'Certificates from Files' which yielded the same errors. Assuming Rancher was somehow getting confused with the certificate I had imported as a secret, I completely wiped the nodes, and re-installed a k8s cluster and then Rancher w/ the 'Rancher Generated Certificates' SSL configuration, but with the same result.

**Environment information**
- Rancher version: v2.2.7
- Helm version: v2.14.3
- node OS: Ubuntu 16.04.6
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): rke installed cluster, per Rancher HA install documentation
- Machine type (cloud/VM/metal) and specifications (CPU/memory): metal: Xeon E3-1220/8GB
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-21T13:09:06Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.5", GitCommit:"0e9fcb426b100a2aea5ed5c25b3d8cfbb01a8acf", GitTreeState:"clean", BuildDate:"2019-08-05T09:13:08Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
docker version
Client: Docker Engine - Community
 Version:           19.03.1
 API version:       1.40
 Go version:        go1.12.5
 Git commit:        74b1e89e8a
 Built:             Thu Jul 25 21:21:35 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.1
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.5
  Git commit:       74b1e89e8a
  Built:            Thu Jul 25 21:20:09 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.6
  GitCommit:        894b81a4b802e4eb2a91d1ce216b8817763c29fb
 runc:
  Version:          1.0.0-rc8
  GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Comment by superseb** [comment]
This sounds identical to https://github.com/rancher/rancher/issues/16697 which has a root cause at https://github.com/rancher/rancher/issues/16697#issuecomment-441659499. Can you check if it is the same root cause?

**Comment by sdanieru** [comment]
@superseb thanks for pointing me in the right direction. There's a lot of conversation around this topic! Yes, it looks like the same problem.
I wanted to find a global, high-level solution to this instead of changing the `ndots:5` default via `dnsConfig` on a per pod basis. What I ended up doing to resolve this (pun intended) was to drop `dns-search my.domain` from `resolv.conf` on my kubernetes host nodes (I was declaring it in `/etc/network/interfaces` since I'm running Ubuntu 16.04). This means I'll need to use FQDNs on those hosts in order to resolve any hosts that are under `my.domain` but that's an acceptable trade-off for me in order to have name resolution 'just work' for all containers in k8s without any config changes.

For anyone who like me needed to better understand how the `ndots:n` and `dns-search` settings interact, I found [this post](https://pracucci.com/kubernetes-dns-resolution-ndots-options-and-why-it-may-affect-application-performances.html) quite helpful. That gave me the basis to understand why this sometimes is causing a problem in kubernetes for certain containers.
